packages/graphql-auth-transformer/src/ModelAuthTransformer.ts

```diff
@@ -294,58 +443,105 @@ Static group authorization should perform as expected.`
+                    const operationName = modelConfiguration.getName('create');
+                    const operationDirectives = this.getDirectivesForRules(rules);
+
+                    if (operationDirectives.length > 0) {
+                        this.addDirectivesToField(ctx, ctx.getMutationTypeName(), operationName, operationDirectives);
+                    }
+                }
+            }
+
             // Break the rules out by strategy.
             const staticGroupAuthorizationRules = this.getStaticGroupRules(rules)
             const dynamicGroupAuthorizationRules = this.getDynamicGroupRules(rules)
             const ownerAuthorizationRules = this.getOwnerRules(rules)
 
-            // Generate the expressions to validate each strategy.
-            const staticGroupAuthorizationExpression = this.resources.staticGroupAuthorizationExpression(staticGroupAuthorizationRules)
+            if (staticGroupAuthorizationRules.length > 0 ||
+                dynamicGroupAuthorizationRules.length > 0 ||
+                ownerAuthorizationRules.length > 0) {
+
+                // Generate the expressions to validate each strategy.
+                const staticGroupAuthorizationExpression = this.resources.staticGroupAuthorizationExpression(staticGroupAuthorizationRules)
+
+                // In create mutations, the dynamic group and ownership authorization checks
+                // are done before calling PutItem.
+                const dynamicGroupAuthorizationExpression = this.resources.dynamicGroupAuthorizationExpressionForCreateOperationsByField(
+                    dynamicGroupAuthorizationRules,
+                    field.name.value
+                )
+                const fieldIsList = (fieldName: string) => {
+                    const field = parent.fields.find(field => field.name.value === fieldName);
+                    if (field) {
+                        return isListType(field.type);
+                    }
+                    return false;
+                }
+                const ownerAuthorizationExpression = this.resources.ownerAuthorizationExpressionForCreateOperationsByField(
+                    ownerAuthorizationRules,
+                    field.name.value,
+                    fieldIsList
+                )
 
-            // In create mutations, the dynamic group and ownership authorization checks
-            // are done before calling PutItem.
-            const dynamicGroupAuthorizationExpression = this.resources.dynamicGroupAuthorizationExpressionForCreateOperationsByField(
-                dynamicGroupAuthorizationRules,
-                field.name.value
-            )
-            const fieldIsList = (fieldName: string) => {
-                const field = parent.fields.find(field => field.name.value === fieldName);
-                if (field) {
-                    return isListType(field.type);
+                const throwIfUnauthorizedExpression = this.resources.throwIfUnauthorized()
```

Comment:
Can you please add some comments throughout this section that give more details and context about what each section is doing? This is getting pretty difficult to decipher and any future devs will appreciate it.

Reply:
Done.